### PR TITLE
[10.0] [MIG] product_customer_code_picking

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Product Customer Code For Stock Picking
=======================================
Based on product_supplierinfo_for_customer, this module loads in every Stock Picking the
customer code defined in the product.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

Commits are already preserved and can you please check the [history](https://github.com/nikul-serpentcs/stock-logistics-workflow/commits/10.0-product_customer_code_picking/product_customer_code_picking).

File: product_customer_code_picking/views/stock_picking_view.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  product customer code picking for OpenERP
+  Copyright (C) 2013 Agile Business Group sagl (<http://www.agilebg.com>).
+    Authors, Nicola Malcontenti, nicola.malcontenti@agilebg.com
+  The licence is in the file __openerp__.py
```
Comment (AaronHForgeFlow) [inline product_customer_code_picking/views/stock_picking_view.xml]: Not really :smile:

**Review by AaronHForgeFlow — COMMENTED** [review]
Code seems ok to me.  Pending functional review. Please change the title from [ADD] to [MIG]

File: product_customer_code_picking/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for move in self:
```
Comment (rousseldenis) [inline product_customer_code_picking/models/stock_picking.py]: Clearer to use something like :

`for move in self.filtered(
lambda m: m.picking_id and m.picking_id.partner_id and m.product_id.supplier_ids):`
    `supplier_info = product_supplierinfo_obj.search('product_tmpl_id', '=', move.product_id.product_tmpl_id.id)`
(...)

File: product_customer_code_picking/README.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======================================
+Product Customer Code For Stock Picking
+=======================================
+Based on product_supplierinfo_for_customer, this module loads in every Stock Picking the
```
Comment (rousseldenis) [inline product_customer_code_picking/README.rst]: To be complete, 'in every Stock Picking **at the move level**...' or something like that to help people to find the info

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Minor comment

**Comment by nikul-serpentcs** [comment]
@rousseldenis @aheficent Could you please review.

File: product_customer_code_picking/models/stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (rousseldenis) [inline product_customer_code_picking/models/stock_picking.py]: As in https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming, please rename file to stock_move.py and change import

**Comment by nikul-serpentcs** [comment]
@rousseldenis I did change a file, Could you please review.

**Review by rousseldenis — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional review on local

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
I was wrong, this is adding the code correctly but the supplier code not the customer code. I'll propose the fix in another PR. Thanks.

File: product_customer_code_picking/models/stock_move.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _get_product_customer_code(self):
```
Comment (rousseldenis) [inline product_customer_code_picking/models/stock_move.py:12]: @nikul-serpentcs Please follow OCA convention on compute name methods (_compute_...).

Moreover, it is mandatory (even if not stored) to decorate the function with depends.
(picking_id, picking_id.partner_id, product_id.supplier_ids, product_tmpl_id.seller_ids.product_code).

Having this done, you can avoid the search in the loop (which is not performant) and get directly the right supplierinfo through move.product_tmpl_id.seller_ids

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by AaronHForgeFlow** [comment]
I created a new PR here #413

**Pull request closed without merging**